xpsparser/parser.py

```diff
@@ -22,42 +22,580 @@
+
 
-from . import metainfo  # pylint: disable=unused-import
-from .metainfo import *
 '''
 This is a test parser for XPS Parser
 '''
 
 logger = logging.getLogger(__name__)
 
 
-class XPSParser(FairdiParser):
+class MetaData:
+    def __init__(self):
+        self.timestamp = ''
+        self.dwell_time = ''
+        self.n_scans = ''
+        self.excitation_energy = ''
+        self.method_type = ''
+        self.data_labels = []
+        self.device_settings = []
+
+class DeviceSettings():
+    def __init__(self):
+        self.device_name = ''
+        self.channel_id = ''
+            
+class AnalyzerSettings(DeviceSettings):
+    def __init__(self):
+        super().__init__()
+        self.pass_energy = ''
+        self.lens_modes = ''
+        self.detector_voltage = ''
+    
+class DataChannel:
+    def __init__(self):   
+        pass
+        
+class MeasurementData:
+    def __init__(self):
+        self.metadata = MetaData()
+        self.data = []
+        
+    def addDataChannel(self, data_channel):
+        self.data += [data_channel]
+            
+
+class ProdigyParserXY():
+    """A parser for reading in ASCII-encoded .xy data from Specs Prodigy.
+    
+    Tested with SpecsLab Prodigy v 4.64.1-r88350.
+    """
+    
+    def __init__(self, **kwargs): 
+        """Construct the parser."""
+        
+        self.kwarg_keys = ['author', 'sample', 'experiment_id', 'project',
+                           'axis_id', 'primary_channel_id']
+        
+        self.default_axis_channel_id = 0
+        self.default_primary_channel_id = 1
+        
+        for key in self.kwarg_keys:
+            if key in kwargs.keys():
+                setattr(self, key, kwargs[key])
+            elif key == 'axis_id':
+                setattr(self, key, self.default_axis_channel_id)
+            elif key == 'primary_channel_id':
+                setattr(self, key, self.default_primary_channel_id) 
+
+        self.primary_spectrum_indicators = ['region', 'Region']
+        self.external_channel_indicators = ['external channel', 'External Channel']
+        
+        self.group_metadata_attribute_map = {
+            'Acquisition Date':'timestamp',
+            'Dwell Time':'dwell_time',
+            'Group':'group_name',
+            'Number of Scans':'n_scans',
+            'Region':'spectrum_region',
+            'Excitation Energy':'excitation_energy',
+            'Values/Curve':'n_values',
+            'Source':'source_label',            
+            }
+        
+        self.settings_attribute_map = {
+            'Analysis Method':'analysis_method',
+            'Analyzer Lens':'analyzer_lens',
+            'Analyzer Slit':'analyzer_slit',
+            'Detector Voltage':'detector_voltage',
+            'Eff. Workfunction':'workfunction',
+            'Scan Mode':'scan_mode',
+            }
+        
+        self.default_energy_unit = 'eV'
+        
+        self.known_channel_labels = {'Ring Current':'ring current',
```

Comment:
Mark's code does not adhere to our style conventions. Dicts should be formatted:
```
self.foo = {
    'key': 'value'
}
```